lib/addons/kube-state-metrics/index.ts

```diff
@@ -0,0 +1,72 @@
+// lib/certmanager_addon.ts
+import { Construct } from 'constructs';
+import merge from "ts-deepmerge";
+import { ClusterInfo, Values } from "../../spi";
+import { createNamespace } from "../../utils";
+import { HelmAddOn, HelmAddOnProps, HelmAddOnUserProps } from "../helm-addon";
+/**
+ * User provided option for the Helm Chart
+ */
+export interface KubeStateMetricsAddOnProps extends HelmAddOnUserProps {
+    /**
+     * To Create Namespace using CDK
+     */    
+    createNamespace?: boolean;
+}
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+const defaultProps: HelmAddOnProps & KubeStateMetricsAddOnProps = {
+  name: "kube-state-metrics",
+  namespace: "kube-system",
+  chart: "kube-state-metrics",
+  version: "4.16.0",

```

Comment:
Please make sure the version is the latest and no warning are produced on the version check.

Reply:
Fixed. Please check in the commit.